pump/server_test.go

```diff
@@ -555,3 +565,222 @@ func (s *listenSuite) TestReturnListener(c *C) {
 	defer l.Close()
 	c.Assert(l, NotNil)
 }
+
+type mockPdCli struct {
+	pd.Client
+}
+
+func (pc *mockPdCli) GetClusterID(ctx context.Context) uint64 {
+	return 8012
+}
+
+func (pc *mockPdCli) Close() {}
+
+type newServerSuite struct {
+	origNewPdCli            func([]string, pd.SecurityOption) (pd.Client, error)
+	origNewKVStore          func(string) (kv.Storage, error)
+	origNewTiKVLockResolver func([]string, config.Security) (*tikv.LockResolver, error)
+	cfg                     *Config
+}
+
+var _ = Suite(&newServerSuite{})
+
+func (s *newServerSuite) SetUpTest(c *C) {
+	s.origNewPdCli = util.NewPdCli
+	s.origNewKVStore = newKVStore
+	s.origNewTiKVLockResolver = newTiKVLockResolver
+
+	// build config
+	etcdClient := testEtcdCluster.RandClient()
+	s.cfg = &Config{
+		ListenAddr:        "http://192.168.199.100:8260",
+		AdvertiseAddr:     "http://192.168.199.100:8260",
+		EtcdURLs:          strings.Join(etcdClient.Endpoints(), ","),
+		DataDir:           "/tmp/pump",
+		HeartbeatInterval: 1500,
+		LogLevel:          "debug",
+		MetricsAddr:       "192.168.199.100:5000",
+		MetricsInterval:   15,
+		Security: security.Config{
+			SSLCA:   "/path/to/ca.pem",
+			SSLCert: "/path/to/drainer.pem",
+			SSLKey:  "/path/to/drainer-key.pem"},
+	}
+}
+
+func (s *newServerSuite) TearDownTest(c *C) {
+	util.NewPdCli = s.origNewPdCli
+	newKVStore = s.origNewKVStore
+	newTiKVLockResolver = s.origNewTiKVLockResolver
+	s.cfg = nil
+}
+
+func (s *newServerSuite) TestCreateNewPumpServerWithInvalidPDClient(c *C) {
+	util.NewPdCli = func([]string, pd.SecurityOption) (pd.Client, error) {
```

Comment:
Store `util.GetPdClient` in a variable to mock it directly so that we don't have to make `NewPdCli` public?

Reply:
Revised